shared/src/commonMain/kotlin/io/customer/shared/tracking/queue/QueueTimer.kt

```diff
@@ -0,0 +1,27 @@
+package io.customer.shared.tracking.queue
+
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.TimeUnit
+
+internal expect fun getQueueTimer(
+    logger: Logger,
+    jobExecutor: JobExecutor,
+): QueueTimer
+
+/**
+ * Wrapper around timer that gives us the ability to provide OS based timers and mocked timers in
+ * tests to make them run faster.
+ */
+internal interface QueueTimer {
```

Comment:
```suggestion
internal interface Timer {
```

Instead of having a timer be specific for the BQ, if it's generic we can use it for more parts of the code.